Odata-docs/client/v6/deal-with-server-driven-paging.md

```diff
@@ -31,3 +31,79 @@ while ((token = response.GetContinuation()) != null)
     response = context.Execute<Person>(token);
 }
 ```
+
+## Nested Pagination
+There are usecases where we need to load pages of entities and also pages of their related entities.
+This example returns related Trips for each Person entity and uses a do�while loop to load Person entities pages and a nested while loop to load pages of related Trips from the data service.
+
+``` csharp
+var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
+DataServiceQueryContinuation<Person> nextLink = null;
+int pageCount = 0;
+int innerPageCount = 0;
+
+try
+{
+    // Execute the query for all people and related trips,
+    // and get the response object.
+    var response =
+        context.People.AddQueryOption("$expand", "Trips")
+        .Execute() as QueryOperationResponse<Person>;
+
+    // With a paged response from the service, use a do...while loop
+    // to enumerate the results before getting the next link.
+    do
+    {
+        // Write the page number.
+        Console.WriteLine("People Page {0}:", ++pageCount);
+
+        // If nextLink is not null, then there is a new page to load.
+        if (nextLink != null)
+        {
+            // Load the new page from the next link URI.
+            response = context.Execute<Person>(nextLink)
+                as QueryOperationResponse<Person>;
+        }
+
+        // Enumerate the People in the response.
+        foreach (Person p in response)
+        {
+            Console.WriteLine("\Person Name: {0}", p.Name);
+            Console.WriteLine("\Trips Page {0}:", ++innerPageCount);
+            // Get the next link for the collection of related trips.
+            DataServiceQueryContinuation<Trip> nextTripsLink =
+                response.GetContinuation(p.Trips);
+
+            foreach (Trip t in p.Trips)
+            {
+                // Print out the trips in the first page.
+                Console.WriteLine("\t\tTripID: {0} - Name: ${1}",
+                    t.TripId, t.Name);
+            }
+
+            while (nextTripsLink != null)
+            {
+                Console.WriteLine("\t\t******Load Trips******");
+
+                // Load the next page of Trips.
+                var tripsResponse = context.LoadProperty(p, "Trips", nextTripsLink);
+                nextTripsLink = tripsResponse.GetContinuation();
```

Comment:
This line fails unless you have it as (I used .NetCore)
```c#
nextTripsLink = tripsResponse.GetContinuation(p.Trips);
```

Reply:
The problem was this line 
`DataServiceQueryContinuation nextTripsLink = response.GetContinuation(p.Trips);`
#Resolved